plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/BaseBigQueryConnectorTest.java

```diff
@@ -840,6 +840,26 @@ protected void verifyColumnNameLengthFailurePermissible(Throwable e)
                 .hasMessageContaining("Fields must contain only letters, numbers, and underscores, start with a letter or underscore, and be at most 300 characters long.");
     }
 
+    @Test
+    public void testCollate()
+    {
+        try (TestTable table = new TestTable(bigQuerySqlExecutor, "test.test_collate", "(col1 STRING)")) {
+            onBigQuery(format("""
+                    INSERT INTO %s
+                        WITH data_values AS (
+                          SELECT *
+                          FROM UNNEST([
+                            COLLATE('B', 'und:ci'),
+                            'b',
+                            'a'
+                          ]) AS character
+                          ORDER BY character)
+                        SELECT * FROM data_values
+                    """, table.getName()));
+            assertQuery(format("SELECT * FROM %s", table.getName()), "VALUES ('a'), ('B'), ('b')");
```

Comment:
This SELECT doesn't test collate at all. You could use `query` table function with outer WHERE condition.

Reply:
The idea here was to test table rows are already ordered because of `COLLATE`, which in this case, because I used `ORDER BY character` during the insert statement, it should have inserted the rows in order 
```
a
B
b
```
and not as
```
B
a
b
```

Reply:
I believe the order during the INSERT doesn't affect SELECT result.

Reply:
Yes, just now verified that the values are just compared irrespective of the order of the result.